server/routes.js

```diff
@@ -14,16 +14,22 @@ const wfClient = new WorkflowClient();
 router.use('/auth', authRoutes);
 
 router.get('/api/namespaces', async function(ctx) {
-  ctx.body = await wfClient.listNamespaces({
-    pageSize: 50,
-    nextPageToken: ctx.query.nextPageToken
-      ? Buffer.from(ctx.query.nextPageToken, 'base64')
-      : undefined,
-  });
+  ctx.body = await wfClient.listNamespaces(
+    {
+      pageSize: 50,
+      nextPageToken: ctx.query.nextPageToken
+        ? Buffer.from(ctx.query.nextPageToken, 'base64')
+        : undefined,
+    },
+    { accessToken: extractAccessToken(ctx) }
```

Comment:
suggestion: the amount of times you are repeating this line of code suggests that there is some kind of middleware that could be used here

Reply:
similar to below,  fully agree this refactoring is needed

Reply:
just to follow up and visibility:created an item in internal backlog to track this